src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -69,15 +65,34 @@ public LogzioSender(String logzioToken, String logzioType, int drainTimeout, int
         queueDirectory = new File(bufferDir);
 
         if (additionalFields != null) {
-            this.additionalFields = Splitter.on(';').omitEmptyStrings().withKeyValueSeparator('=').split(additionalFields);
+
+            JsonObject logMessage = formatMessageAsJson(new Date().getTime(), "Level", "Message", "Logger", "Thread");
+
+            Splitter.on(';').omitEmptyStrings().withKeyValueSeparator('=').split(additionalFields).forEach((k, v) -> {
+
+                if (logMessage.get(k) != null) {
+                    reporter.warning("You requested to add value name " + k + ", but it is reserved. Ignoring it.");
+                }
+                else {
+                    if (v.startsWith("$")) {
+                        String environmentValue = System.getenv(v.replace("$", ""));
+                        if (environmentValue != null) {
+                            additionalFieldsMap.put(k, environmentValue);
+                        }
+                    } else {
+                        additionalFieldsMap.put(k, v);
+                    }
+                }
+            });
         }
 
         try {
             if (addHostname) {
-                hostname = InetAddress.getLocalHost().getHostName();
+                String hostname = InetAddress.getLocalHost().getHostName();
+                additionalFieldsMap.put("hostname", hostname);
             }
         } catch (UnknownHostException e) {
-            reporter.error("Could not resolve host! ignoring it..", e);
+            reporter.warning("Could not resolve host! ignoring it..", e);
```

Comment:
Maybe "The configuration addHostName was specified but the host could not be resolved, thus the field 'hostname' will not be added"